sdks/python/apache_beam/runners/sdf_utils.py

```diff
@@ -0,0 +1,173 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pytype: skip-file
+
+"""Common utility class to help SDK harness to execute an SDF. """
+
+from __future__ import absolute_import
+from __future__ import division
+
+import logging
+import threading
+from builtins import object
+from collections import namedtuple
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Optional
+from typing import Tuple
+
+from apache_beam.utils import timestamp
+
+if TYPE_CHECKING:
+  from apache_beam.io.iobase import RestrictionTracker
+  from apache_beam.utils.timestamp import Timestamp
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SplitResultPrimary = namedtuple(
+    'SplitResultPrimary', 'windowed_value')
+
+SplitResultResidual = namedtuple(
+    'SplitResultResidual',
+    'windowed_value current_watermark deferred_timestamp')
```

Comment:
please use `typing.NamedTuple` so that we can track the types of members.

Reply:
I'm not sure whether I can use `typing.NamedTuple` directly since it is only supported since py3.5

Reply:
You can do so safely.  `typing` is a requirement for python versions less than 3.8 (and there's a [PR to adjust that to 3.5](https://github.com/apache/beam/pull/10821)).  We're already importing typing in numerous places throughout the code.

Reply:
Changed in https://github.com/apache/beam/pull/10802/commits/8aa9821439fbb941c83c61e34b52aedc1404dacc